apps/chat/src/components/AppsEditor/GeneralInfoView/form.ts

```diff
@@ -0,0 +1,204 @@
+import { Path, RegisterOptions } from 'react-hook-form';
+
+import { safeStringifyApplicationFeatures } from '@/src/utils/app/application';
+import { notAllowedSymbols } from '@/src/utils/app/file';
+import { getNextDefaultName } from '@/src/utils/app/folders';
+import { ApiUtils } from '@/src/utils/server/api';
+
+import { ApiDetailedApplicationTypeSchema } from '@/src/types/application-type-schema';
+import {
+  ApiApplicationResponseDefault,
+  ApplicationStatus,
+  ApplicationType,
+  CustomApplicationModel,
+} from '@/src/types/applications';
+import { EntityType } from '@/src/types/common';
+import { QuickAppConfig } from '@/src/types/quick-apps';
+
+import {
+  FEATURES_ENDPOINTS,
+  FEATURES_ENDPOINTS_DEFAULT_VALUES,
+  FEATURES_ENDPOINTS_NAMES,
+} from '@/src/constants/applications';
+import { DEFAULT_APPLICATION_NAME } from '@/src/constants/default-ui-settings';
+import { DEFAULT_VERSION } from '@/src/constants/public';
+
+import { DynamicField } from '../../Common/Forms/DynamicFormFields';
+
+import { ShareEntity } from '@epam/ai-dial-shared';
+
+export interface ApplicationGeneralInfoFormData {
+  name: string;
+  version: string;
+  iconUrl: string;
+  description: string;
+  topics: string[];
+  completionUrl: string;
+  id: string;
+  reference: string;
+  applicationProperties?: Record<string, unknown> | QuickAppConfig | null;
+  inputAttachmentTypes?: string[];
+  maxInputAttachments?: number;
+  features?: string | null;
+  env?: DynamicField[];
+  sources?: string;
+  sourceFiles?: string[];
+  runtime?: string;
+  endpoints?: DynamicField[];
+  functionStatus?: ApplicationStatus;
+}
+
+type Options<T extends Path<ApplicationGeneralInfoFormData>> = Omit<
+  RegisterOptions<ApplicationGeneralInfoFormData, T>,
+  'disabled' | 'valueAsNumber' | 'valueAsDate'
+>;
+
+type Validators = {
+  [K in keyof ApplicationGeneralInfoFormData]?: Options<K>;
+};
+
+export const getDefaultValues = (
+  applicationData?: ApiApplicationResponseDefault,
+  bucket?: string,
+  pythonVersion?: string,
+  models?: ShareEntity[],
+): ApplicationGeneralInfoFormData => {
+  return {
+    name:
+      applicationData?.display_name ??
+      getNextDefaultName(DEFAULT_APPLICATION_NAME, models ?? [], 0, true),
```

Comment:
Need to dispatch getModels on page load, otherwise we will get error on trying to create an application with default name if we open apps-editor page without navigating to it from marketplace

<img width="885" alt="Снимок экрана 2025-02-13 в 15 15 42" src="https://github.com/user-attachments/assets/c4896511-3cf3-4c4a-86d5-5e92dd43f484" />

Reply:
fixed